PostgreSQL - SQL error when search for module

alikon opened the pull request:
#### Steps to reproduce the issue

Go to the Extension -> Module Manager in Backend  and  search for a term
#### Expected result

If some match are found the result module list is returned
#### Actual result

Got an ERROR  SQL QUERY SYNTAX ERROR  REGEXP
#### System information

PostgreSQL 9.3.5
Joomla 3.4.0
#### Additional comments

the `REGEXP` operator work only for Mysql  
for Postgres we should use  `~*`

fix #6159

alikon committed:
fix 6159

mbabker commented:
It looks like this is going to fail as you added the method to the query classes yet you're calling it from the driver class.

Also, we need a method in the base driver/query class (whichever it is supposed to be in) otherwise it'll need to be wrapped in a `method_exists()` check as I don't see a method here for SQL Server.

alikon committed:
damned autocomplete

alikon commented:
@mbabker you are right my fault,  hope better now

for the SQL server  based on this http://www.sql-workbench.net/dbms_comparison.html we are in trouble

so i ask  why we need a posix pattern matching in module/template/style search ?
for db's portability should be better the trivial LIKE

mbabker commented:
Can't say I have an answer for that one, but if it works the same with a LIKE then go for it.

alikon committed:
switched to like

alikon committed:
make travis happy

alikon committed:
make travis happy with spaces

alikon committed:
travis not a friend of mine

alikon commented:
i've switched the 3 models from REGEXP to LIKE

despite i don't have a clue for MSSQL  posix pattern matching i've added the "portability" method regexp for Mysql and Postgresql  for future use

gunjanpatel commented on libraries/joomla/database/query/mysqli.php:
```diff
@@ -101,4 +101,23 @@ public function setLimit($limit = 0, $offset = 0)
+	 *
+	 * @since   11.3
+	 */
+	public function regexp($value)
```
Good work @alikon 
You have not used this functions anywhere. So, this doesn't need anymore or it's for future purpose?

gunjanpatel replied:
and I think you should add function here `https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/database/driver/postgresql.php` if you want to use it like, `$db->regex()` in query. Similarly, `$db->escape()`

alikon commented:
It's for future purpose 
The Regexp is an operator and usually it's used mainly in  where condition or this is the way I think at
So I've putted  that method in the query class

alikon commented:
@gunjanpatel 
as usual my typos in the comment  i've writed
```
* Usage:
* $query->where('field ' . $db->regexp($search));
```
instead of
```
$query->where('field ' . $query->regexp($search));
```

gunjanpatel commented:
Super @alikon Thanks.

alikon committed:
fixed docs

fixed  usage comment

alikon committed:
fixed docs

fixed usage comment

ketchupmonki commented:
Successfully tested in postgre.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6188">issues.joomla.org/joomla-cms/6188</a>.</sub>

waader commented:
@test works with postgresql, mssql, mysql! Thanks @alikon!

It also fixes search in template manager for postgresql and mssql.

zero-24 commented:
Moving to RTC! Thanks for coding and testing!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6188">issues.joomla.org/joomla-cms/6188</a>.</sub>

Kubik-Rubik commented:
Great work @alikon!
